File: 0701-0800/0754-Reach a Number/0754-Reach a Number.cpp
class Solution {
public:
    int reachNumber(int target) {
        target = abs(target);        
        int sum = 0;
        int result = 0;
        int i = 0;
        while (sum < target) {
            sum += ++i;
        }
        
        if ((sum - target) % 2 == 0) {
            result = i; 
        }
        else if (i % 2 == 0){
            result = i + 1;
        }
        else {
            result = i + 2;
        }
        
        return result;
    }
};

File: 0701-0800/0754-Reach a Number/0754-Reach a Number.py
class Solution:
    def reachNumber(self, target: int) -> int:
        target = abs(target)
        i = 0
        while target > 0:
            i += 1
            target -= i
        if target % 2 == 0:
            result = i
        elif i % 2 == 0:
            result = i + 1
        else:
            result = i + 2
        return result
